### ISSUE TITLE
'Unnecessary' hint fades out only first token

### ISSUE THREAD

**Issue by chrmarti** [Microsoft/vscode#53099]
Issue Type: <b>Bug</b>

Testing #52218

The 'unnecessary' hint spans the entire document, only the first token is faded:
![image](https://user-images.githubusercontent.com/9205389/41962167-a2de58f0-79f4-11e8-96e5-b9b8fb55b662.png)

VS Code version: Code - Insiders 1.25.0-insider (f2fa6bea46bde9856e6a593e4d4c5af2b303ac66, 2018-06-27T05:11:02.122Z)
OS version: Darwin x64 17.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4770HQ CPU @ 2.20GHz (8 x 2200)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (1.15GB free)|
|Process Argv|/Applications/Visual Studio Code - Insiders.app/Contents/MacOS/Electron --extensions-dir /Users/chrmarti/Development/repos/tmp/ext --user-data-dir /Users/chrmarti/Development/repos/tmp/user|
|Screen Reader|no|
|VM|20%|

</details>Extensions: none

**Comment by Gama11** [comment]
Are there plans to address this in the near future (since it's not assigned to a milestone or tagged as a bug)?

I ran into this when trying to use the new API to fade out inactive conditional compilation blocks in the Haxe extension.

![](https://i.imgur.com/X9qWYHJ.png)

I noticed that if the severity is != `Hint`, the fading also works properly across the entire range:

![](https://i.imgur.com/f2Wyofz.png)

Of course, that leaves you with some undesirable squiggly lines, so it's not very useful in practice - but maybe it helps with finding the source of the bug.
____

Some test code I wrote to isolate it (before finding this issue):

```ts
import * as vscode from "vscode";

export function activate(context: vscode.ExtensionContext) {
	var diagnostics = vscode.languages.createDiagnosticCollection("foo");
	vscode.window.onDidChangeActiveTextEditor(editor => {
		if (editor != null) {
			var doc = editor.document;
			var range = new vscode.Range(new vscode.Position(0, 0), new vscode.Position(doc.lineCount, doc.lineAt(doc.lineCount - 1).lineNumber));
			var diagnostic = new vscode.Diagnostic(range, "Inactive section", vscode.DiagnosticSeverity.Hint);
			diagnostic.tags = [vscode.DiagnosticTag.Unnecessary];
			diagnostics.set(editor.document.uri, [diagnostic]);
		}
	});
}
```

![](https://i.imgur.com/uDf20Af.png)

**Comment by mjbvz** [comment]
Fixed by e659ffd6acf00cb7201f5bebc1e8431f4530c93c
